src/core/attributeformmodelbase.cpp

```diff
@@ -194,6 +200,13 @@ QgsAttributeEditorContainer *AttributeFormModelBase::generateRootContainer() con
       root->addChildElement( field );
     }
   }
+  //get relations
+  QList<QgsRelation> referencingRelations = QgsProject::instance()->relationManager()->referencedRelations( mLayer );
+  for ( int i = 0; i < referencingRelations.size(); ++i )
```

Comment:
Range based for loop. const QList one line up and for ( const Relation& relation : referencingRelations ) here